src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -553,8 +553,26 @@ internal static List<CompletionResult> CompleteCommandParameter(CompletionContex
                 partialName = string.Empty;
             }
 
+            bool bindPositionalParameters = true;
+            // If the user tries to tab complete a new parameter in front of a positional argument like: ls -<Tab> C:\
+            // the user may want to add the parameter name so we don't want to bind positional arguments
+            if (commandAst is not null && string.Equals(context.TokenAtCursor?.Text, "-", StringComparison.OrdinalIgnoreCase))
```

Comment:
I see code above (lines 545-543) already detect dash. So questions are (1) do we really need this specific detection here, (2) if need should it be in block above?

Reply:
Good catch, I've moved it inside that block now.